extensions/airport/src/index.tsx

```diff
@@ -0,0 +1,151 @@
+import { ActionPanel, CopyToClipboardAction, List, OpenInBrowserAction, showToast, ToastStyle } from "@raycast/api";
+import { useState, useEffect } from "react";
+import fetch from "node-fetch";
+
+interface App {
+  id: string;
+  name: string;
+  icon: string;
+  categories: string;
+  description: string;
+  status: string;
+  platform: string;
+}
+
+interface Apps {
+  all: App[];
+  featured: any[];
+  popular: any[];
+  new: any[];
+  upcoming: any[];
+  waitlist: any[];
+}
+interface Response {
+  apps: Apps;
+  categories: any[];
+  sections: any[];
+}
+
+export default function ArticleList() {
+  const [state, setState] = useState<{
+    all: App[] | any;
+    popular: string[] | any;
+    featured: string[] | any;
+    new: string[] | any;
+    upcoming: string[] | any;
+    waitlist: string[] | any;
+  }>({ all: [], popular: [], featured: [], new: [], upcoming: [], waitlist: [] });
+
+  useEffect(() => {
+    async function fetch() {
+      showToast(ToastStyle.Animated, "Loading Apps");
+      const appsResponse = await fetchApps();
+      setState((oldState) => ({
+        ...oldState,
+        all: appsResponse?.apps.all.filter(
+          (app) =>
+            !appsResponse?.apps.popular.includes(app.id) &&
+            !appsResponse?.apps.featured.includes(app.id) &&
+            !appsResponse?.apps.new.includes(app.id) &&
+            !appsResponse?.apps.upcoming.includes(app.id) &&
+            !appsResponse?.apps.waitlist.includes(app.id)
+        ),
+        popular: appsResponse?.apps.all.filter((app) => appsResponse?.apps.popular.includes(app.id)),
+        featured: appsResponse?.apps.all.filter((app) => appsResponse?.apps.featured.includes(app.id)),
+        new: appsResponse?.apps.all.filter(
+          (app) =>
+            !appsResponse?.apps.popular.includes(app.id) &&
+            !appsResponse?.apps.featured.includes(app.id) &&
+            appsResponse?.apps.new.includes(app.id) &&
+            !appsResponse?.apps.upcoming.includes(app.id) &&
+            !appsResponse?.apps.waitlist.includes(app.id)
+        ),
+        upcoming: appsResponse?.apps.all.filter(
+          (app) =>
+            !appsResponse?.apps.popular.includes(app.id) &&
+            !appsResponse?.apps.featured.includes(app.id) &&
+            !appsResponse?.apps.new.includes(app.id) &&
+            appsResponse?.apps.upcoming.includes(app.id) &&
+            !appsResponse?.apps.waitlist.includes(app.id)
+        ),
+        waitlist: appsResponse?.apps.all.filter(
+          (app) =>
+            !appsResponse?.apps.popular.includes(app.id) &&
+            !appsResponse?.apps.featured.includes(app.id) &&
+            !appsResponse?.apps.new.includes(app.id) &&
+            !appsResponse?.apps.upcoming.includes(app.id) &&
+            appsResponse?.apps.waitlist.includes(app.id)
+        ),
+      }));
+      showToast(ToastStyle.Success, "Fetched Latest Apps");
```

Comment:
It's enough to show the loading indicator in the searchbar. You don't need to add the toast. More info [here](https://developers.raycast.com/information/best-practices#show-loading-indicator).

```suggestion
      const appsResponse = await fetchApps();
      setState((oldState) => ({
        ...oldState,
        all: appsResponse?.apps.all.filter(
          (app) =>
            !appsResponse?.apps.popular.includes(app.id) &&
            !appsResponse?.apps.featured.includes(app.id) &&
            !appsResponse?.apps.new.includes(app.id) &&
            !appsResponse?.apps.upcoming.includes(app.id) &&
            !appsResponse?.apps.waitlist.includes(app.id)
        ),
        popular: appsResponse?.apps.all.filter((app) => appsResponse?.apps.popular.includes(app.id)),
        featured: appsResponse?.apps.all.filter((app) => appsResponse?.apps.featured.includes(app.id)),
        new: appsResponse?.apps.all.filter(
          (app) =>
            !appsResponse?.apps.popular.includes(app.id) &&
            !appsResponse?.apps.featured.includes(app.id) &&
            appsResponse?.apps.new.includes(app.id) &&
            !appsResponse?.apps.upcoming.includes(app.id) &&
            !appsResponse?.apps.waitlist.includes(app.id)
        ),
        upcoming: appsResponse?.apps.all.filter(
          (app) =>
            !appsResponse?.apps.popular.includes(app.id) &&
            !appsResponse?.apps.featured.includes(app.id) &&
            !appsResponse?.apps.new.includes(app.id) &&
            appsResponse?.apps.upcoming.includes(app.id) &&
            !appsResponse?.apps.waitlist.includes(app.id)
        ),
        waitlist: appsResponse?.apps.all.filter(
          (app) =>
            !appsResponse?.apps.popular.includes(app.id) &&
            !appsResponse?.apps.featured.includes(app.id) &&
            !appsResponse?.apps.new.includes(app.id) &&
            !appsResponse?.apps.upcoming.includes(app.id) &&
            appsResponse?.apps.waitlist.includes(app.id)
        ),
      }));
```